### DISCUSSION TITLE
Question about preventing abuse on single Dagster instance when using dagster-k8s/config

### DISCUSSION THREAD

**Discussion by dagsir** [dagster-io/dagster#17758] [Q&A]
Another question: We have several independent teams using a single Dagster instance. We are using the `K8sRunLauncher` and each team uses the `dagster-k8s/config` tag to configure their jobs to launch in a dedicated K8s namespace for their team. This works as expected but it seems to raise multiple security concerns. The `dagster-k8s/config` essentially allows any configuration a user wants, including, but not limited to, loading additional secrets from the k8s environment, attaching additional volumes or even requesting root privileges on a K8s node for the container running the job. What are the recommended best security practices when using `dagster-k8s/config`? <https://docs.dagster.io/deployment/guides/kubernetes/customizing-your-deployment#security|The documentation> mentions using "a ServiceAccount bound to a properly scoped Role to launch Jobs and create other Kubernetes resources". The issue with that is that when multiple teams are using a single Dagster instance (and thereby a single Run Launcher), there doesn't seem to be a way to completely prevent abuse apart from somehow disabling `dagster-k8s/config` altogether. Any thoughts on that or recommendations?

---

*The question was originally asked in [Dagster Slack](https://dagster.slack.com/archives/C01U954MEER/p1699337995128549?thread_ts=1699337995.128549&cid=C01U954MEER).*

**Comment by gibsondan** [comment]
One solution to this would be for us to add a configuration option to the K8sRunLauncher that restricts which fields can be set on dagster-k8s/config.
